src/components/text-editor/text-editor.pw.tsx

```diff
@@ -0,0 +1,613 @@
+    await mount(<TextEditorCustomValidation />);
+
+    const textInput = textEditorInput(page);
+    await textInput.focus();
+    await getComponent(page, "icon").first().hover();
+    await expect(getDataElementByValue(page, "tooltip")).toHaveText(
+      "There is an error"
+    );
+
+    expect(
+      await getStyle(getDataElementByValue(page, "error"), "color", "before")
+    ).toBe(VALIDATION.ERROR);
+
+    const iconColor = await page.evaluate(() => {
+      const validationIcon = document.querySelector(`[data-element="error"]`);
+      if (!validationIcon) {
+        return null;
+      }
+      const beforePseudoElement = window.getComputedStyle(
+        validationIcon,
+        "::before"
+      );
+      return beforePseudoElement ? beforePseudoElement.color : null;
+    });
+    expect(iconColor).toBe(VALIDATION.ERROR);
+  });
+
+  test(`should render with warning validation state`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<TextEditorCustomValidation />);
+
+    const textInput = textEditorInput(page);
+    await textInput.focus();
+    for (let i = 0; i < 5; i++) {
+      await page.keyboard.press("Delete");
+    }
+
+    await getComponent(page, "icon").first().hover();
+    await expect(getDataElementByValue(page, "tooltip")).toHaveText(
+      "There is a warning"
+    );
+
+    expect(
+      await getStyle(getDataElementByValue(page, "warning"), "color", "before")
+    ).toBe(VALIDATION.WARNING);
+
+    const iconColor = await page.evaluate(() => {
+      const validationIcon = document.querySelector(`[data-element="warning"]`);
+      if (!validationIcon) {
+        return null;
+      }
+      const beforePseudoElement = window.getComputedStyle(
+        validationIcon,
+        "::before"
+      );
+      return beforePseudoElement ? beforePseudoElement.color : null;
+    });
+    expect(iconColor).toBe(VALIDATION.WARNING);
+  });
+
+  test(`should render with info validation state`, async ({ mount, page }) => {
+    await mount(<TextEditorCustomValidation />);
+
+    const textInput = textEditorInput(page);
+    await textInput.focus();
+    for (let i = 0; i < 10; i++) {
+      await page.keyboard.press("Delete");
+    }
+
+    await getComponent(page, "icon").first().hover();
+    await expect(getDataElementByValue(page, "tooltip")).toHaveText(
+      "There is an info"
+    );
+
+    expect(
+      await getStyle(getDataElementByValue(page, "info"), "color", "before")
+    ).toBe(VALIDATION.INFO);
+
+    const iconColor = await page.evaluate(() => {
+      const validationIcon = document.querySelector(`[data-element="info"]`);
+      if (!validationIcon) {
+        return null;
+      }
+      const beforePseudoElement = window.getComputedStyle(
+        validationIcon,
+        "::before"
+      );
+      return beforePseudoElement ? beforePseudoElement.color : null;
+    });
+    expect(iconColor).toBe(VALIDATION.INFO);
+  });
+
```

Comment:
I think it would be useful to also have a test for the input hint text prop in the new validation example using the `testdata` chars.

Reply:
This is also tested in unit tests so I'm not overly concerned about it being tested in a browse. If you want to add it for the sake of consistency with the other pw files then that makes sense, but these type of tests will be unit tests in the long run though

Reply:
Does the unit test cover the special char strings that we use in Playwright tests? If so, then I'm happy to leave it in unit test. The more that can be shifted left while providing full coverage, the better for me.

Reply:
No it just tests that it is configurable, ie I pass foo and label is foo type thing. Is there any reason why we need to test these like this as if they render or not isn't down to the component but browser/ project that determines it? As I say don't mind it being added just not sure there's any increased value beyond consistency

Reply:
In the past, we had issues where some special characters in labels, hint fields etc, didn't perform the correct escape action and littered them with a load of `&` chars instead of the correct ones. I remember it regressed at one point, so we tend to test all of these in the browser now.